plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -286,61 +290,65 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
     public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
-        org.apache.iceberg.Table icebergTable = getIcebergTable(tableOperationsProvider, session, table.getSchemaTableName());
-
-        TupleDomain<IcebergColumnHandle> enforcedPredicate = table.getEnforcedPredicate();
 
         if (table.getSnapshotId().isEmpty()) {
             // A table with missing snapshot id produces no splits, so we optimize here by returning
             // TupleDomain.none() as the predicate
             return new ConnectorTableProperties(TupleDomain.none(), Optional.empty(), Optional.empty(), Optional.empty(), ImmutableList.of());
         }
 
-        TableScan tableScan = icebergTable.newScan()
-                .useSnapshot(table.getSnapshotId().get())
-                .filter(toIcebergExpression(enforcedPredicate))
-                .includeColumnStats();
-
-        CloseableIterable<FileScanTask> files = tableScan.planFiles();
+        org.apache.iceberg.Table icebergTable = getIcebergTable(session, table.getSchemaTableName());
 
         // Extract identity partition fields that are present in all partition specs, for creating the discrete predicates.
         Set<Integer> partitionSourceIds = identityPartitionColumnsInAllSpecs(icebergTable);
 
+        TupleDomain<IcebergColumnHandle> enforcedPredicate = table.getEnforcedPredicate();
+
         DiscretePredicates discretePredicates = null;
         if (!partitionSourceIds.isEmpty()) {
             // Extract identity partition columns
             Map<Integer, IcebergColumnHandle> columns = getColumns(icebergTable.schema(), typeManager).stream()
                     .filter(column -> partitionSourceIds.contains(column.getId()))
                     .collect(toImmutableMap(IcebergColumnHandle::getId, Function.identity()));
 
-            Iterable<TupleDomain<ColumnHandle>> discreteTupleDomain = Iterables.transform(files, fileScan -> {
```

Comment:
I think we can make the diff shorter if we keep `files`:
```java
Iterable<FileScanTask> files = ...
```
Also, a few things to consider here

1. We might want to ensure thread safety of the returned object
2. Computing the value is expensive, so we should do it only once
3. The original is a `CloseableIterator`, so we should close it to avoid leaks (existing bug)

I think we can address these by doing
```java
Supplier<Iterable<FileScanTask>> lazyFiles = Suppliers.memoize(() -> {
    try (CloseableIterable<FileScanTask> iterator = tableScan.planFiles()) {
        return ImmutableList.copyOf(iterator);
    }
    catch (IOException e) {
        throw new UncheckedIOException(e);
    }
});
Iterable<FileScanTask> files = () -> lazyFiles.get().iterator();